src/lucky/cookies/cookie_jar.cr

```diff
@@ -159,10 +162,22 @@ class Lucky::CookieJar
     end
   end
 
+  private def update_from_legacy_value(value : String) : String
+    decoded_value = URI.decode_www_form(value)
+    LUCKY_ENCRYPTION_PREFIX + decoded_value.lchop(LEGACY_LUCKY_ENCRYPTION_PREFIX)
+  end
+
   private def encrypted_with_lucky?(value : String) : Bool
     value.starts_with?(LUCKY_ENCRYPTION_PREFIX)
   end
 
+  # legacy encrypted values had a \n between the encoded lucky and -- and were also www form encoded
+  # this allows apps made before 0.27.0 to not have to log all users out
+  private def encrypted_with_legacy?(value : String) : Bool
+    decoded_value = URI.decode_www_form(value)
+    decoded_value.starts_with?(LEGACY_LUCKY_ENCRYPTION_PREFIX)
+  end
```

Comment:
I think this is fine, but what's a good upgrade path we can take so we can drop this by the next release or something? Maybe we can just add a deprecation notice, as well as some info logging? It'll be noisy, but only happens while people are on legacy cookies. Then for the next release, we yank this stuff out.

Reply:
I don't think there is a deprecation notice that would be useful for devs. It's not their app that needs to update, it's their users. For instance, if a user doesn't sign in while the app's deployed as 0.27 and we remove this code in 0.28 they'll run into this issue as well if their session cookie still exists. It would also be a problem if devs skip the release with the legacy code

Reply:
Ok, maybe we keep it until 1.0, but for 1.0 we rip it out. It'll be a breaking change, but it'll be up to the developers to have figured out on their end how to get their users to clear the cookies or whatever. I think we should definitely at least log a warning letting people know.

Reply:
To be clear, this change will fix users going forward. This reads them in the old way but writes cookies the new way so all users will be fixed going forward. The only worry is apps skipping this legacy handling or users somehow keeping their cookies and not logging on while the legacy version is deployed.

Neither of those seem like warnings we can give the devs, do they?

Reply:
What's the failure mode when this code doesn't exist?

Reply:
Tokens encrypted the old way cause this error https://github.com/luckyframework/lucky/blob/9474f6d1d996dc73dab813a1e1af12edd97d5afb/src/lucky/cookies/cookie_jar.cr#L151-L158